src/service/expenses.service.js

```diff
@@ -0,0 +1,129 @@
+// eslint-disable-next-line no-unused-vars
+const { Expenses } = require('../model/expenses.model');
+
+const { getUserById } = require('./user.service');
+
+/**
+ * @type {Expenses[]}
+ */
+let expenses = [];
+
+const getExpenses = (expensesURLParams) => {
+  const { userId, categories, from, to } = expensesURLParams;
+
+  let filteredExpenses = [...expenses];
+
+  if (userId) {
+    filteredExpenses = filteredExpenses.filter(
+      (expense) => expense.userId === Number(userId),
+    );
+  }
+
+  if (categories) {
+    filteredExpenses = filteredExpenses.filter(
+      (expense) => expense.category === categories,
+    );
+  }
+
+  if (from) {
+    filteredExpenses = filteredExpenses.filter(
+      (expense) => expense.spentAt >= from,
+    );
+  }
+
+  if (to) {
+    filteredExpenses = filteredExpenses.filter(
+      (expense) => expense.spentAt <= to,
+    );
+  }
+
+  return filteredExpenses;
+};
+
+const createExpense = (expense) => {
+  const userId = expense.userId;
+  const title = expense.title;
+  const amount = expense.amount;
+  const category = expense.category;
+  const note = expense.note;
+  const spentAt = expense.spentAt;
+
+  let newExpense;
+
+  if (!userId || !title || !amount || !category) {
+    return null;
+  }
+
+  const findUser = getUserById(userId);
+
+  if (!findUser) {
+    return undefined;
+  }
+
+  if (expenses.length !== 0) {
+    newExpense = {
+      id: expenses[expenses.length - 1].id + 1,
```

Comment:
Seems like you can just use:
`expenses.length + 1`

Reply:
i couldn't do this lolool. Let's imagine that a have id's 1, 2, 3, 4, 5. If I delete id '3' the array length will be four(4) and the entities that stayed are 1, 2, 4, 5. Then, if I create another entity, the id number will be five(5) because array length is 4, but there is already exists an entity with id 5 you know?

I thought about it and I chose to get the last entity id and add plus one based on it.

`array[array.length - 1].id + 1`